tests/components/nsw_fuel_station/test_sensor.py

```diff
@@ -72,39 +70,33 @@ def get_reference_data(self):
         )
 
 
-class TestNSWFuelStation(unittest.TestCase):
-    """Test the NSW Fuel Station sensor platform."""
-
-    def setUp(self):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self.config = VALID_CONFIG
-        self.addCleanup(self.hass.stop)
-
-    @patch(
-        "homeassistant.components.nsw_fuel_station.sensor.FuelCheckClient",
-        new=FuelCheckClientMock,
-    )
-    def test_setup(self):
-        """Test the setup with custom settings."""
-        with assert_setup_component(1, sensor.DOMAIN):
-            assert setup_component(self.hass, sensor.DOMAIN, {"sensor": VALID_CONFIG})
-            self.hass.block_till_done()
-
-        fake_entities = ["my_fake_station_p95", "my_fake_station_e10"]
-
-        for entity_id in fake_entities:
-            state = self.hass.states.get(f"sensor.{entity_id}")
-            assert state is not None
-
-    @patch(
-        "homeassistant.components.nsw_fuel_station.sensor.FuelCheckClient",
-        new=FuelCheckClientMock,
-    )
-    def test_sensor_values(self):
-        """Test retrieval of sensor values."""
-        assert setup_component(self.hass, sensor.DOMAIN, {"sensor": VALID_CONFIG})
-        self.hass.block_till_done()
-
-        assert "140.0" == self.hass.states.get("sensor.my_fake_station_e10").state
-        assert "150.0" == self.hass.states.get("sensor.my_fake_station_p95").state
+@patch(
+    "homeassistant.components.nsw_fuel_station.sensor.FuelCheckClient",
+    new=FuelCheckClientMock,
+)
+async def test_setup(hass):
+    """Test the setup with custom settings."""
+    with assert_setup_component(1, sensor.DOMAIN):
+        assert await async_setup_component(
+            hass, sensor.DOMAIN, {"sensor": VALID_CONFIG}
+        )
+        await hass.async_block_till_done()
+
+    fake_entities = ["my_fake_station_p95", "my_fake_station_e10"]
+
+    for entity_id in fake_entities:
+        state = hass.states.get(f"sensor.{entity_id}")
+        assert state is not None
+
+
+@patch(
+    "homeassistant.components.nsw_fuel_station.sensor.FuelCheckClient",
+    new=FuelCheckClientMock,
+)
+async def test_sensor_values(hass):
+    """Test retrieval of sensor values."""
+    assert await async_setup_component(hass, sensor.DOMAIN, {"sensor": VALID_CONFIG})
+    await hass.async_block_till_done()
+
+    assert "140.0" == hass.states.get("sensor.my_fake_station_e10").state
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6356951)